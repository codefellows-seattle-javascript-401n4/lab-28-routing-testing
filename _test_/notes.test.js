import React from 'react';
import renderer from 'react-test-renderer';
import Jest from 'jest';
import NoteCreateForm from '../src/components/noteCreateForm.js';
import uuid from 'uuid/v1';


test('create a new note with the content pizza, editing false, and completed true', () => {
  let content = 'pizza';
  let editing = false;
  let completed = true;
  let createNote = (state) => {
    expect(NoteCreateForm.state().content).toBe('pizza');
    expect(NoteCreateForm.state().editing).toBe(false);
    expect(NoteCreateForm.state().completed).toBe(true);
  };

  const component = renderer.create(
    <NoteCreateForm />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('should fail because of change in content', () => {
  const component = renderer.create(
    <NoteCreateForm content=''/>
  );

  const other = renderer.create(
    <NoteCreateForm content='sushi'/>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(other).toBe(tree);
});

test('should pass with unique ids',() => {
  const component = renderer.create(
    <NoteCreateForm id='2222'/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('should fail with non unique ids', () => {
  const component = renderer.create(
    <NoteCreateForm id='2222'/>
  );
  const other = renderer.create(
    <NoteCreateForm id='2222'/>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(other).toBe(tree);

});
