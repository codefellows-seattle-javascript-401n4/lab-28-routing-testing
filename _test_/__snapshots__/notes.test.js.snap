// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create a new note with the content and editing false 1`] = `
<div
  className="inputForm"
>
  <form
    onSubmit={[Function]}
  >
    <label>
       
      <h4>
        Enter Your Note 
      </h4>
      <input
        className="noteForm"
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Submit!
      </button>
    </label>
  </form>
</div>
`;

exports[`create a new note with the content pizza, editing false, and completed true 1`] = `
<div
  className="inputForm"
>
  <form
    onSubmit={[Function]}
  >
    <label>
       
      <h4>
        Enter Your Note 
      </h4>
      <input
        className="noteForm"
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Submit!
      </button>
    </label>
  </form>
</div>
`;

exports[`should fail because of change in content 1`] = `
<div
  className="inputForm"
>
  <form
    onSubmit={[Function]}
  >
    <label>
       
      <h4>
        Enter Your Note 
      </h4>
      <input
        className="noteForm"
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Submit!
      </button>
    </label>
  </form>
</div>
`;

exports[`should fail with non unique ids 1`] = `
<div
  className="inputForm"
>
  <form
    onSubmit={[Function]}
  >
    <label>
       
      <h4>
        Enter Your Note 
      </h4>
      <input
        className="noteForm"
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Submit!
      </button>
    </label>
  </form>
</div>
`;

exports[`should fail with non unique ids 2`] = `
<div
  className="inputForm"
>
  <form
    onSubmit={[Function]}
  >
    <label>
       
      <h4>
        Enter Your Note 
      </h4>
      <input
        className="noteForm"
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Submit!
      </button>
    </label>
  </form>
</div>
`;

exports[`should pass with unique ids 1`] = `
<div
  className="inputForm"
>
  <form
    onSubmit={[Function]}
  >
    <label>
       
      <h4>
        Enter Your Note 
      </h4>
      <input
        className="noteForm"
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Submit!
      </button>
    </label>
  </form>
</div>
`;
